### ISSUE TITLE
Embark attack cities

### ISSUE THREAD

**Issue by Enrico-Swagolo** [LoneGazebo/Community-Patch-DLL#3191]
_1. Mod version (i.e Date - 4/23):_

current
_2. Mod list (if using Vox Populi only, leave blank):_

_3. Error description:_ 
basically I've built a city on a hill and have a great general in it (increasing CS). Isabella keeps on embarking Warriors and attacking the city, killing her warriors off in the process.

In addition, the bug of enemies attacking barbie camps as a welcome is still in, but it's minor.

**Comment by LoneGazebo** [comment]
@ilteroi is this worth investigating?

**Comment by ilteroi** [comment]
do you have screenshots? it's hard to judge the situations (both of them actually) from this terse description

**Comment by Enrico-Swagolo** [comment]
I could see if I have autosaves going this far, but I will make a bigger description instead -

I've declared war on Spain as Poland. I have a coastal city on a hill with a horseman and Great General in it (18CS). At this point, Isabella has tons and tons of warriors (she's been spamming them forever it seems). She's on the same continent but can only reach me through water - so she embarks. No problems here. Instead of landing her soldiers on land next to the city or anywhere else in my lands, she just attacks the city with warriors which instantly kills them (18CS city vs 5CS warrior embarked likely without amphibious = dead), Fun fact - her archers she embarks instead, to the tiles near the city as she probably should have done with warriors. I assume it's because it's impossible to attack cities with embarked ranged troops. 

While it doesn't really matter that much as I had spammed Horsemen anyway and they can oneshot her Warriors, if it was actual good units she was throwing away, it'd be much worse. It could be some sort of a pathfinding problem, too.

About the second, just like in the previous version I often see AI attack a barbie camp as soon as they see it, whether they use Scouts (forcing the AI to heal) or Warrior but it's almost always Scouts that fall victim to that. It always happens if the unit has spare movement, if they notice the camp as they lose all the movement and are unable to reach it - they are very likely to not attack later, next turn.
